website/blog/deployers.md

```diff
@@ -0,0 +1,592 @@
+# How to Implement a Service Weaver Deployer
+
+<div class="blog-author">Michael Whittaker</div>
+<div class="blog-date">TODO, 2023</div>
+
+Service Weaver allows you to deploy an application binary in many different
+ways. For example, you can deploy an application in a [single
+process][singleprocess], across [multiple processes][multiprocess], or in [the
+cloud][gke]. The code that deploys a Service Weaver application is called,
+unsurprisingly, a **deployer**. In this blog post, we provide a high-level
+overview of *what* deployers are and then get down to the nitty-gritty of
+exactly *how* they work, culminating in us implementing one completely from
+scratch. This blog post assumes you have a basic familiarity with how to write
+Service Weaver applications. If you don't, we recommend you read the
+[step-by-step tutorial][tutorial].
+
+## Weavelets
+
+To understand deployers, we must first understand weavelets. A Service Weaver
+application is compiled into a single executable binary. The Service Weaver
+libraries linked in to the binary include a small agent called a **weavelet**,
+which is created when you call [weaver.Init][Init]. A weavelet's main
+responsibility is to start and manage a set of [components][].
+
+When a Service Weaver application is deployed, there isn't just one weavelet. If
+there were, Service Weaver applications wouldn't be very distributed. Rather,
+deployers run your binary multiple times&mdash;in different processes across
+different machines&mdash;to launch multiple weavelets which work together to
+execute your distributed application.
+
+Every weavelet hosts a potentially different set of components. Because
+components are replicated, a component may be hosted by multiple weavelets. For
+example, consider an application with components `A`, `B`, and `C`. An
+example deployment consisting of three weavelets is shown in the figure below.
+
+<figure>
+  <img src="deployers/assets/weavelets.svg" alt="TODO">
+</figure>
+
+Weavelet 1 hosts components `A` and `B`; weavelet 2 hosts components `B` and
+`C`, and weavelet 3 hosts component `C`. You'll also notice that every weavelet
+has a unique network address. When a component hosted by weavelet `i` invokes a
+remote method on a component, weavelet `i` selects a replica of the component
+hosted on some weavelet `j` to perform the RPC. When weavelet `j` receives the
+RPC, it dispatches the request to its locally running replica of the component
+and responds to weavelet `i`. In this way, weavelets are all interconnected.
+
+## Deployers
+
+A deployer has three main responsibilities: components, listeners, and
```

Comment:
I think this intro is missing something that states that a deployer manages multiple weavelets (kind of like the point you made earlier about if it's a single weavelet, things wouldn't be distributed)

Reply:
Done. The new Overview section at the top of the blog might also make this clearer.